pubtools/_quay/signature_handler.py

```diff
@@ -378,6 +368,37 @@ def upload_signatures_to_pyxis(self, claim_mesages, signature_messages, max_item
                 env_vars,
             )
 
+    def remove_signatures_from_pyxis(self, signatures_to_remove):
+        """
+        Upload signatures to Pyxis by using a pubtools-pyxis entrypoint.
+
+        Args:
+            signature_to_remove ([repo, tag, manifest]):
+                tuples of signature composed keys which need to be removed
+        """
+        LOG.info("Removing outdated signatures from pyxis")
+
+        for sig_to_remove in signatures_to_remove:
+            repo, digest, reference = sig_to_remove
+            args = [
+                "--pyxis-server",
+                self.target_settings["pyxis_server"],
+                "--pyxis-krb-principal",
+                self.target_settings["iib_krb_principal"],
+            ]
+            if "iib_krb_ktfile" in self.target_settings:
+                args += ["--pyxis-krb-ktfile", self.target_settings["iib_krb_ktfile"]]
+            args += ["--repo", repo, "--digest", digest, "--reference", reference]
```

Comment:
Could you check PR https://github.com/release-engineering/pubtools-pyxis/pull/20/files ? The signatures search supports the same inputs as get_signatures entrypoint. I would suggest using just digest to match to-be-removed signatures.

Although now that I think about it more, it's possible that an image with the same digest was reuploaded, generating no new signatures. In this case it would be undesirable to remove these signatures.

Perhaps the right approach is to get existing signatures by digest (using https://github.com/release-engineering/pubtools-pyxis/pull/18), check them against the result of construct_item_claim_messages, and only remove those which were not re-added. To-be-deleted signatures would be referenced by pyxis's internal ID. This means that I would have to change my PR https://github.com/release-engineering/pubtools-pyxis/pull/20 to accept signature IDs

Have I missed something? What do you think?

Reply:
I think if I simply remove only signature which are not in manifest_claims it should be enough. And combination of manifest digest + reference should be unique enough.